engine/platform/win32/con_win.c

```diff
@@ -568,17 +581,25 @@ destroy win32 console
 */
 void Wcon_DestroyConsole( void )
 {
+	int return_key = VK_RETURN;
+
 	// last text message into console or log
 	Con_Reportf( "Sys_FreeLibrary: Unloading xash.dll\n" );
 
 	Sys_CloseLog();
 
-	if( s_wcd.hWnd )
+	if( !s_wcd.is_attached && s_wcd.hWnd )
 	{
 		ShowWindow( s_wcd.hWnd, SW_HIDE );
 		s_wcd.hWnd = 0;
 	}
 
+	if( s_wcd.is_attached )
+	{
+		SetConsoleTitle( &s_wcd.old_title );
```

Comment:
You can minimize the scope of return_key variable by defining it's inside a block here, just at the beginning.

Also, I highly doubt that the third parameter of SendMessage is actually just int*. Is it even supposed to be a pointer? MSDN says wParam is just a character code for WM_CHAR.

Reply:
WPARAM defined as UINT_PTR or uintptr_t.

Reply:
I'm not a Windows programmer you know but everywhere I see, WM_CHAR uses just a key value rather than a pointer. Even in the Wine source!

Reply:
Even in the ReactOS %)
`_PTR` suffix means different size on different architectures, not a pointer.

Reply:
I'm talking about the code above, which sends a pointer to a variable located in the stack.

Something really tells me it's just wrong.

Reply:
I thought for a long time and it also seemed wrong to me. But if you remove the pointer, then not a return is sent, but some ^M. The last parameter should be NULL, not 0, but this does not affect anything

Reply:
^M is carriage return. As I read in Discord chat, it's only to solve the "missing" command prompt?

How does the pointer get decoded then?

Reply:
Try to send just '\n'.
VK_RETURN has the same code as CR in ascii, but code of LF is reserved.
Pointer cause undefined behavior anyway.

Reply:
@nekonomicon 

We have two different behaviors in cmd and powershell.

In cmd, it is enough to send VK_RETURN, and the prompt will be displayed immediately after the log. Nice and neat

Some kind of mess in the powershell. The prompt is drawn over the text when any key is pressed. If I send '\n' as you suggested, then the caret advances to the next line, displaying '>>'. Sending VK_RETURN will display '^M'...

Reply:
As a last resort, you can print "Press Enter to continue". xD

Reply:
@SNMetamorph well, maybe you can help solve the mystery of should wParam be a pointer or a key code, in this discussion.

Or get rid of this nonsense code, which again looks like a workaround of buggy Windows interactive shells and send the other parts.

That's why I always tell PRs should be minimal and solve exactly one problem, and leave debatable stuff to the next patches. Or at least next commit lol.

Reply:
@a1batross I think it's better to get rid of SendMessage and just try to print `\n` with conventional methods.